cli/js/io.ts

```diff
@@ -110,3 +112,46 @@ export function* iterSync(
     yield b.subarray(0, result);
   }
 }
+
+export function toWriter(
+  streamWriter: WritableStreamDefaultWriter<Uint8Array>
+): Writer {
+  return {
+    async write(p: Uint8Array): Promise<number> {
+      await streamWriter.ready;
+      await streamWriter.write(p);
+      return p.length;
+    },
+  };
+}
+
+export function toReader(
+  streamReader: ReadableStreamDefaultReader<Uint8Array | string>
```

Comment:
Why are we handling strings here? The encoding is tacked on.

Reply:
You're right, got confused, good catch.

Reply:
Actually, the `ReadableStreamsDefaultController` allows you to enqueue any javascript type, so I think we should handle strings too. Check  `toReaderStringCheck`.

So we should handle strings IMO.

Reply:
```javascript
const chunks = [undefined, null, "", "deno"];

const readableStream = new ReadableStream({
	pull(controller) {
		if (chunks.length === 0) return controller.close();
		const chunk = chunks.shift();

		controller.enqueue(chunk);
	},
});

const reader = readableStream.getReader();

while (true) {
	const res = await reader.read();
	if (res.done) break;

	console.log(`Chunk: ${typeof res.value}`);
}
```

```
Chunk: undefined
Chunk: object
Chunk: string
Chunk: string
```

Reply:
Yes... I can see it handles any type but the Deno IO interfaces only handle `Uint8Array`s so we can only bridge from `ReadableStreamDefaultReader<Uint8Array>` to `Deno.Reader`. Of course you can "support more types" by allowing them, checking for them, and converting them to `Uint8Array`s. But that can be done by the user.

Handling strings here is completely arbitrary and encoding them the way you are is presumptuous. In your example, the user would have to encode it themselves (to their own specification!) here: https://github.com/denoland/deno/blob/3454b9cbc3d975cd2ab071998f612e789c4b88cc/cli/tests/unit/io_test.ts#L138-L142

Reply:
If we're exposing the API, I think it should support as many types as we can. Most of the things we do (`iter`, `copy`...) can be done by the user, but we still do them, so don't agree with your logic.

In any case, I don't mind removing `string`, and leaving only `Uint8Array`. I can create a third-party module that can handle strings.

Reply:
> If we're exposing the API, I think it should support as many types as we can.

So every type? You can support any type by adding some arbitrary encoding logic for it, like you've done for strings. We do this to support common use cases in `ServerRequest.respond()`, for example. Not for low level IO interfaces. Most of the feedback/changes this PR has had so far were in the name of being "minimal".

Reply:
Pushed the changes. I still think it should support `utf-8` strings but can do it in a third-party lib.